src/matter/fabric/FabricManager.ts

```diff
@@ -18,6 +18,10 @@ export class FabricManager {
         return new FabricIndex(this.fabrics.length);
     }
 
+    removeFabric(fabricIndex: FabricIndex) {
+        this.fabrics.splice(this.fabrics.findIndex(fabric => fabric.fabricIndex.index === fabricIndex.index), 1);
```

Comment:
Here you assume that the the array idex is the same as fabric Index, right? can we do this?

Maybe better to re-set fabrict with a "filtered" array where yoiu just remove the entry with the relevant index?